sdk/storage/azure-storage-files-datalake/src/datalake_file_system_client.cpp

```diff
@@ -248,6 +248,7 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
       const ListPathsOptions& options) const
   {
     Details::DataLakeRestClient::FileSystem::ListPathsOptions protocolLayerOptions;
+    protocolLayerOptions.Resource = Models::FileSystemResourceType::Filesystem;
```

Comment:
Why don't we need to set this before?

Why not `FileSystem`?

Reply:
It is due to the enum not being parsed as an optional value, due to it being an enum with one possible value. It was parsed as a constant.
However, to be future compatible, I don't think what used to be done is correct here, so I'm making it an enum.
The enum's string value is `filesystem`, I don't think it makes sense to call it `FileSystem`

Reply:
> The enum's string value is filesystem, I don't think it makes sense to call it FileSystem

why not?

Reply:
As talked offline, this is due to weird service behavior of this query parameter being called:
```
    "FileSystemResource": {
	  "name": "resource",
	  "in": "query",
	  "description": "The value must be \"filesystem\" for all filesystem operations.",
	  "required": true,
	  "type": "string",
	  "enum": [
		"filesystem"
	  ],
	  "x-ms-enum": {
		"name": "FileSystemResourceType",
		"modelAsString": false
	  }
	},
```
While for other query parameter:
```
          {
            "name": "action",
            "in": "query",
            "description": "Optional. If the value is \"getStatus\" only the system defined properties for the path are returned. If the value is \"getAccessControl\" the access control list is returned in the response headers (Hierarchical Namespace must be enabled for the account), otherwise the properties are returned.",
            "required": false,
            "type": "string",
            "enum": [
              "getAccessControl",
              "getStatus"
            ],
            "x-ms-enum": {
              "name": "PathGetPropertiesAction",
              "modelAsString": false
            }
          },
```
Service itself is introducing this inconsistency, there is little I can do in client.